RUN3/AliAnalysisTaskAO2Dconverter.cxx

```diff
@@ -2551,8 +2553,20 @@ AliAnalysisTaskAO2Dconverter::FwdTrackPars AliAnalysisTaskAO2Dconverter::MUONtoF
   convertedTrack.fChi2MatchMCHMID = AliMathBase::TruncateFloatFraction(MUONTrack.GetChi2MatchTrigger(), mMuonTrCov);
   convertedTrack.fRAtAbsorberEnd = AliMathBase::TruncateFloatFraction(MUONTrack.GetRAtAbsorberEnd(), mMuonTrCov);
   convertedTrack.fPDca = AliMathBase::TruncateFloatFraction(pdca, mMuonTrCov);
-  
+
+  //MCH and MID words
   convertedTrack.fMCHBitMap = MUONTrack.GetMuonClusterMap();
+  UInt_t midpattern = MUONTrack.GetHitsPatternInTrigChTrk();
+  UShort_t midbitmap = 0;
+  for (int icath=0; icath < 2; icath++){
+    for (int ich=0; ich < 4; ich++){
+      if (AliESDMuonTrack::IsChamberHit(midpattern,icath,ich)) midbitmap |= 1<<(ich+icath*4);
+    }
+  }
+  convertedTrack.fMIDBitMap = midbitmap;
+  UChar_t midboard = (AliESDMuonTrack::GetCrossedBoard(midpattern) & 0xFF) ;
+  convertedTrack.fMIDBoards = ((midboard << 24) | (midboard << 16) | (midboard  << 8) | midboard) & 0xFFFFFFFF;
```

Comment:
Hi @nicolovalle ,
I'm not 100% sure about this line. You defined the midboard as a UChar_t, meaning that the compiler knows it has 8 bytes. Now, if you do midboard<<24, I think that the compiler still thinks that the result of the operation is a UChar_t...and therefore it cuts whatever it finds after bit 8. This means that you will always get 0.
I think it would be better to do:
`UInt_t middboard = static_cast<UInt_t>(AliESDMuonTrack::GetCrossedBoard(midpattern));`
(notice that 0xFF is done already by GetCrossedBoard: no need to do it again).
And then:
`convertedTrack.fMIDBoards = ((midboard << 24) | (midboard << 16) | (midboard << 8) | midboard);`
(notice again that 0xFFFFFFFF should not be necessary)

Reply:
Hi @dstocco , 
thanks for having checked this. Actually the code works (`MIDBoards` is filled properly).. maybe an automatic cast is done during the bit shift? 
In any case what you say is correct, so I'm going to make the change.